src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputOperationConverter.cs

```diff
@@ -85,17 +87,38 @@ public static InputOperation CreateOperation(ref Utf8JsonReader reader, string?
                 Console.Error.WriteLine($"[Warn]: InputOperation '{name}' does not have a description");
                 description = $"{name.Humanize()}.";
             }
+            crossLanguageDefinitionId = crossLanguageDefinitionId ?? throw new JsonException("InputOperation must have crossLanguageDefinitionId");
             uri = uri ?? throw new JsonException("InputOperation must have uri");
             path = path ?? throw new JsonException("InputOperation must have path");
             parameters = parameters ?? throw new JsonException("InputOperation must have parameters");
             responses = responses ?? throw new JsonException("InputOperation must have responses");
 
-            var inputOperation = new InputOperation(name, resourceName, summary, deprecated, description, accessibility, parameters, responses, httpMethod, requestBodyMediaType, uri, path, externalDocsUri, requestMediaTypes, bufferResponse, longRunning, paging, generateProtocolMethod, generateConvenienceMethod, keepClientDefaultValue);
+            var inputOperation = new InputOperation(name, resourceName, summary, deprecated, description, accessibility, parameters, responses, httpMethod, requestBodyMediaType, uri, path, externalDocsUri, requestMediaTypes, bufferResponse, longRunning, paging, generateProtocolMethod, generateConvenienceMethod, crossLanguageDefinitionId, keepClientDefaultValue)
+            {
+                IsNameChanged = IsNameChanged(crossLanguageDefinitionId, name)
+            };
             if (id != null)
             {
                 resolver.AddReference(id, inputOperation);
             }
             return inputOperation;
+
+            static bool IsNameChanged(string crossLanguageDefinitionId, string name)
+            {
+                var span = crossLanguageDefinitionId.AsSpan();
+                int lastDotIndex = span.LastIndexOf('.');
+                if (lastDotIndex < 0)
```

Comment:
crosssLanguageDefinitionId of an operation should be {service-name-space}.{operation-name}. So it seems that this condition should not exist. Right? If the scenario exists, and we filter out this, we may need to change the method name `IsNameChanged` because in this secario, even the name is changed, it will return `false`. Then the method name is not consistent with its real functionality.

Reply:
Let me add an assertion here to throw when it does not have the dot.